### PR TITLE
Add more options to conformance script

### PR DESCRIPTION
**Pull request by smirnov-alexey**
### Details:
 - *item1*
 - *...*

### Tickets:
 - *ticket-id*

### CODE REVIEW

**Comment by smirnov-alexey** [comment]
@ArtemySkrebkov @iefode fyi

**Comment by smirnov-alexey** [comment]
How do I update wiki page? https://github.com/openvinotoolkit/openvino/wiki/Conformance-tests#api-conformance

**Comment by smirnov-alexey** [comment]
@ArtemySkrebkov @iefode please, review and let's merge it

File: src/tests/ie_test_utils/functional_test_utils/layer_tests_summary/run_conformance.py [hunk 1]
```diff
@@ -88,13 +87,17 @@ def parse_arguments():
...
+    nworkers_help = "Specify number of workers to run in parallel. Set to -1 to utilize all threads. The default value is 1"
+    gtest_filter_helper = "Specify gtest filter to apply when running test. E.g. *Add*:*BinaryConv*. The default value is None"
     dump_conformance_help = "Set '1' if you want to create Conformance IRs from custom/downloaded models. In other cases, set 0. The default value is '1'"
 
     parser.add_argument("-m", "--models_path", help=models_path_help, type=str, required=False, default=NO_MODEL_CONSTANT)
     parser.add_argument("-d", "--device", help= device_help, type=str, required=False, default="CPU")
     parser.add_argument("-ov", "--ov_path", help=ov_help, type=str, required=False, default=get_ov_path())
     parser.add_argument("-w", "--working_dir", help=working_dir_help, type=str, required=False, default=get_default_working_dir())
     parser.add_argument("-t", "--type", help=type_help, type=str, required=False, default="OP")
+    parser.add_argument("-nw", "--num_workers", help=nworkers_help, type=int, required=False, default=1)
```
Comment (iefode) [inline src/tests/ie_test_utils/functional_test_utils/layer_tests_summary/run_conformance.py:99]: `-w`, `--workers`. Moreover Use the same as for gtest parallel. Default value should be `CPU count -1`

File: src/tests/ie_test_utils/functional_test_utils/layer_tests_summary/run_conformance.py [hunk 2]
```diff
@@ -88,13 +87,17 @@ def parse_arguments():
...
+    parser.add_argument("-gtf", "--gtest_filter", help=gtest_filter_helper, type=str, required=False, default=None)
```
Comment (iefode) [inline src/tests/ie_test_utils/functional_test_utils/layer_tests_summary/run_conformance.py:100]: do not use short command line option, it's not clear. Lets use `--gtest_filter`

File: src/tests/ie_test_utils/functional_test_utils/layer_tests_summary/run_conformance.py [hunk 3]
```diff
@@ -117,6 +120,10 @@ def __init__(self, device:str, model_path:os.path, ov_path:os.path, type:str, wo
             logger.error(f"Incorrect conformance type: {type}. Please use 'OP' or 'API'")
             exit(-1)
         self._type = type
+        self._num_workers = num_workers
+        if not gtest_filter:
+            gtest_filter = "**"
```
Comment (iefode) [inline src/tests/ie_test_utils/functional_test_utils/layer_tests_summary/run_conformance.py:125]: "*"

File: src/tests/ie_test_utils/functional_test_utils/layer_tests_summary/run_conformance.py [hunk 4]
```diff
@@ -228,9 +235,11 @@ def _prepare_filelist(self):
 
     def run_conformance(self):
         gtest_parallel_path = os.path.join(self.__download_repo(GTEST_PARALLEL_URL, GTEST_PARALLEL_BRANCH), "thirdparty", "gtest-parallel", "gtest_parallel.py")
-        worker_num = os.cpu_count()
-        if worker_num > 2:
-            worker_num = worker_num - 1
+
+        if self._num_workers == -1:
```
Comment (iefode) [inline src/tests/ie_test_utils/functional_test_utils/layer_tests_summary/run_conformance.py:239]: remove this logic, Use `cpu count -1` as default value

File: src/tests/ie_test_utils/functional_test_utils/layer_tests_summary/run_conformance.py [hunk 5]
```diff
@@ -289,6 +299,8 @@ def start_pipeline(self, dump_models: bool):
         logger.info(f"[ARGUMENTS] --models_path = {self._model_path}")
         logger.info(f"[ARGUMENTS] --working_dir = {self._working_dir}")
         logger.info(f"[ARGUMENTS] --type = {self._type}")
+        logger.info(f"[ARGUMENTS] --num_workers = {self._num_workers}")
```
Comment (iefode) [inline src/tests/ie_test_utils/functional_test_utils/layer_tests_summary/run_conformance.py:302]: --workers

**Review by iefode — CHANGES_REQUESTED** [review]

File: src/tests/ie_test_utils/functional_test_utils/layer_tests_summary/run_conformance.py (hunk 2 above)
Comment (smirnov-alexey) [reply src/tests/ie_test_utils/functional_test_utils/layer_tests_summary/run_conformance.py]: Done

File: src/tests/ie_test_utils/functional_test_utils/layer_tests_summary/run_conformance.py (hunk 3 above)
Comment (smirnov-alexey) [reply src/tests/ie_test_utils/functional_test_utils/layer_tests_summary/run_conformance.py]: Done

File: src/tests/ie_test_utils/functional_test_utils/layer_tests_summary/run_conformance.py (hunk 4 above)
Comment (smirnov-alexey) [reply src/tests/ie_test_utils/functional_test_utils/layer_tests_summary/run_conformance.py]: Done

File: src/tests/ie_test_utils/functional_test_utils/layer_tests_summary/run_conformance.py (hunk 5 above)
Comment (smirnov-alexey) [reply src/tests/ie_test_utils/functional_test_utils/layer_tests_summary/run_conformance.py]: Done

**Review by iefode — APPROVED** [review]

**Review by sbalandi — APPROVED** [review]

**Review by ArtemySkrebkov — APPROVED** [review]

**Pull request merged**